### PR TITLE
[F] Create Connection Page

### LINKED ISSUE #3: [F] Create new connection
After the user presses the "Create New Connection" button, they are taken to an input screen where they can input their connection details.

AC:
- navigates to another prompt (screen / popup) to enter details 
- passes info to database functions

### PR DESCRIPTION
**Pull request by alex-hobby**
# Description

Adds a new page with a form to input connection info for a new connection.

Form:
```
Nickname
Type (DBProvider enum values)
ConnectionConfig: {
  Address
  Port
  Username
  Password
} or {
  Connection String 
}
Remember (not supported currently)
```

Error handling:
- Reminder if field is left blank
- Database must have been connected, otherwise a modal pops up
- Only navigates to execute screen if all checks pass

## Link to the issue you are closing

Fixes #3 

## Screenshot or other testing that you have completed

https://user-images.githubusercontent.com/29695042/193064586-3b182024-c5d9-49e8-809b-0feaf4558df7.mov

# Checklist:

- [x] ESLint passes
- [x] Prettier passes
- [x] Included comments where necessary
- [x] Updated README if needed
- [x] Verified that code runs and generates no errors/warnings

### CODE REVIEW

File: src/renderer/components/NewConnectionForm.tsx [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        password: form.password,
+      },
+    };
+
+    try {
+      // creates the server connection in sqlite and connects to it
+      await window.connections.ipcRenderer.create(connection);
+
+      // check if server exists by attempting to fetch procedures - might be a better way to do this
+      await window.procedures.ipcRenderer.fetchProcedures();
```
Comment (alex-hobby) [inline src/renderer/components/NewConnectionForm.tsx:117]: @adityavidyadharan what is the best way to check that the server exists? the create function does not seem to return any errors for garbage data

File: src/renderer/components/NewConnectionForm.tsx [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+          )}
+          {connectionStringActive && (
+            <Form.Group className="mb-2" controlId="connectionString">
+              <Form.Label>Connection String</Form.Label>
+              <Form.Control
+                className="form-control-sm"
+                value={form.connectionString}
+                onChange={(e) => setField('connectionString', e.target.value)}
+                type="text"
+                placeholder="Ex: postgresql://username:password@address:port"
```
Comment (alex-hobby) [inline src/renderer/components/NewConnectionForm.tsx:258]: Highlighting the example text that I'm unsure about

Comment (adityavidyadharan) [reply src/renderer/components/NewConnectionForm.tsx:258]: It's of the format `postgresql://dbuser:secretpassword@database.server.com:3211/mydb`

**Review by adityavidyadharan — APPROVED** [review]
Changes look good besides the linting issues

**Comment by alex-hobby** [comment]
> Changes look good besides the linting issues

Not really sure how to get around that... technically the fields are guaranteed to be non-null where those linting errors occur due to the structure of the conditional logic, but I have to do the non-null assertion because typescript isn't smart enough to know that.

**Review by Ruben-agredano — APPROVED** [review]
Nice Design. 
The error handling is a good idea as well.

**Pull request merged**